scripts/SREP/hard-prune-image-registry/script.sh

```diff
@@ -0,0 +1,115 @@
+                _echo_to_stderr "Try: \"oc patch configs.imageregistry.operator.openshift.io/cluster --patch='{\"spec\":{\"readOnly\":false}}' --type=merge\""
+        fi
+    fi
+    
+    # No need to run if the service account was never set or retrieved properly
+    if [[ -n "${SERVICE_ACCOUNT}" ]]
+    then
+        # Remove the image-pruner role from the service account
+        echo "Removing the image-pruner role from the service account..."
+        if ! oc adm policy remove-cluster-role-from-user system:image-pruner -z "${SERVICE_ACCOUNT}" -n "${NAMESPACE}"
+        then
+            _echo_to_stderr "WARNING: Failed to remove the image-pruner role from the service account."
+            _echo_to_stderr "Try: \"oc adm policy remove-cluster-role-from-user system:image-pruner -z ${SERVICE_ACCOUNT} -n ${NAMESPACE}\""
+        fi
+    fi
+    
+    # On the off chance that someone in the future uses the cleanup function before the end of the script
+    set -o errexit
+
+    # If any failures have occured, make it explicit at the end of the script for manual follow up
+    if [[ "${FAILED}" != "false" ]]
+    then
+        _echo_to_stderr "FAILURE: Pruning the image registry has failed.  Please check the output above for more details."
+    fi
+}
+
+# _delete_orphaned_blobs runs the actual command inside one of the pods from the registry deployment
+_delete_orphaned_blobs() {
+    local registry_pod # SC2155
+    registry_pod=$(oc get -n "${NAMESPACE}" pod -l docker-registry=default -o jsonpath='{.items[0].metadata.name}')
+    oc -n "${NAMESPACE}" exec pod/"${registry_pod}" -- /bin/sh -c '/usr/bin/dockerregistry -prune=delete'
+}
+
+# _retry 
+_retry() {
+    local n=1
+    local max=3
+
+    while true
+    do
+      "${@}" && break
+      if [[ $n -lt $max ]]
+      then
+          _echo_to_stderr "Command failed. Attempt $n/$max:"
+          ((n++))
+      else
+          _echo_to_stderr "The command has failed after $n attempts."
+          FAILED=true
+          break
+      fi
+    done
+}
+
+main() {
+    # Get the service asccount for the registry deployment
+    echo "Looking up the service account for the image registry deployment..."
+    SERVICE_ACCOUNT=$(oc get -n "${NAMESPACE}" -o jsonpath='{.spec.template.spec.serviceAccountName}' deploy/image-registry)
+
+    echo "Adding the image-pruner role to the service account..."
+    oc adm policy add-cluster-role-to-user system:image-pruner -z "${SERVICE_ACCOUNT}" -n ${NAMESPACE}
+    
+    # Mark the image registry read-only
+    echo "Marking the image registry read-only..."
+    oc patch configs.imageregistry.operator.openshift.io/cluster -p '{"spec":{"readOnly":true}}' --type=merge
+    
+    # Delete the orphaned blobs (delete mode: `--prune=delete`)
+    # This command can be run manually with `--prune=check` to see what would be deleted:
+    # EG: oc -n ${NAMESPACE} exec pod/${REGISTRY_POD} -- /bin/sh -c 'REGISTRY_LOG_LEVEL=info /usr/bin/dockerregistry -prune=check'
```

Comment:
Do we still want these comments? I'm thinking no

Reply:
I added those for folks who might be following up, and considered adding a flag for the check command.  Fair point though, will remove.